pkg/sql/sem/eval/parse_doid.go

```diff
@@ -178,12 +179,54 @@ func ParseDOid(ctx context.Context, evalCtx *Context, s string, t *types.T) (*tr
 		if err != nil {
 			return nil, err
 		}
-		id, err := evalCtx.Planner.ResolveTableName(ctx, &tn)
+
+		if tn.ExplicitCatalog {
```

Comment:
Can you encapsulate this new logic? It's not very simple and we'll want to comment it.

Reply:
done